examples/cudax/simple_p2p/simple_p2p.cu

```diff
@@ -0,0 +1,253 @@
+    cuda::mr::pinned_memory_resource(), dev0_buffer.size());
+  std::generate(host_buffer.begin(), host_buffer.end(), []() {
+    static int i = 0;
+    return (i++) % 4096;
+  });
+
+  cudax::copy_bytes(dev0_stream, host_buffer, dev0_buffer);
+  dev1_stream.wait(dev0_stream);
+
+  // Kernel launch configuration
+  auto dims = cudax::distribute<512>(dev0_buffer.size());
+
+  // Run kernel on GPU 1, reading input from the GPU 0 buffer, writing output to the GPU 1 buffer
+  printf("Run kernel on GPU%d, taking source data from GPU%d and writing to "
+         "GPU%d...\n",
+         dev1.get(),
+         dev0.get(),
+         dev1.get());
+  cudax::launch(dev1_stream, dims, simple_kernel{}, dev0_buffer, dev1_buffer);
+  dev0_stream.wait(dev1_stream);
+
+  // Run kernel on GPU 0, reading input from the GPU 1 buffer, writing output to the GPU 0 buffer
+  printf("Run kernel on GPU%d, taking source data from GPU%d and writing to "
+         "GPU%d...\n",
+         dev0.get(),
+         dev1.get(),
+         dev0.get());
+  cudax::launch(dev0_stream, dims, simple_kernel{}, dev1_buffer, dev0_buffer);
+
+  // Copy data back to host and verify
+  printf("Copy data back to host from GPU%d and verify results...\n", dev0.get());
+  cudax::copy_bytes(dev0_stream, dev0_buffer, host_buffer);
+  dev0_stream.wait();
+
+  int error_count = 0;
+  for (int i = 0; i < host_buffer.size(); i++)
+  {
+    cuda::std::span host_span(host_buffer);
+    // Re-generate input data and apply 2x '* 2.0f' computation of both kernel runs
+    float expected = float(i % 4096) * 2.0f * 2.0f;
+    if (host_span[i] != expected)
+    {
+      printf("Verification error @ element %i: val = %f, ref = %f\n", i, host_span[i], expected);
+
+      if (error_count++ > 10)
+      {
+        break;
+      }
+    }
+  }
+  if (error_count != 0)
+  {
+    printf("Test failed!\n");
+    exit(EXIT_FAILURE);
+  }
+}
+
+int main(int argc, char** argv)
+try
+{
+  const int test_waived = 2;
+  printf("[%s] - Starting...\n", argv[0]);
+
+  // Number of GPUs
+  printf("Checking for multiple GPUs...\n");
+  printf("CUDA-capable device count: %lu\n", cudax::devices.size());
+
+  if (cudax::devices.size() < 2)
+  {
+    printf("Two or more GPUs with Peer-to-Peer access capability are required for %s.\n", argv[0]);
+    printf("Waiving test.\n");
+    exit(test_waived);
+  }
+
+  auto peers = find_peers_group();
+
+  if (peers.size() == 0)
+  {
+    printf("Two or more GPUs with Peer-to-Peer access capability are required, waving the test.\n");
+    exit(test_waived);
+  }
+
+  cudax::stream dev0_stream(peers[0]);
+  cudax::stream dev1_stream(peers[1]);
+
+  printf("Enabling peer access between GPU%d and GPU%d...\n", peers[0].get(), peers[1].get());
+  cudax::mr::async_memory_resource dev0_resource(peers[0]);
+  dev0_resource.enable_peer_access(peers[1]);
+  cudax::mr::async_memory_resource dev1_resource(peers[1]);
+  dev1_resource.enable_peer_access(peers[0]);
```

Comment:
My understanding was that because its done on memory pool / memory resource level, we say (this memory).enable_peer_access(device_to_access it).

The other way does not make much sense semantically, but if its not obvious we could thing about some more explicit name

Reply:
what do you think of `mr.enable_access_from(dev1)` ?